config/config-api/src/main/java/com/thoughtworks/go/config/MailHost.java

```diff
@@ -221,20 +223,23 @@ private void resetPassword(String password) {
     }
 
     private void setPasswordIfNotBlank(String password) {
-        if (StringUtils.isBlank(password)) {
+        this.password = StringUtils.stripToNull(password);
+        this.encryptedPassword = StringUtils.stripToNull(encryptedPassword);
```

Comment:
Is this necessary? Since we are reassigning the encrypted value later?

Reply:
Yes this is necessary.

For a situation where a user submits an empty string for `password` and `encryptedPassword`, you'd need to `null`ify both before doing anything.